program/src/processor.rs

```diff
@@ -29,12 +29,16 @@ use {
 // its a gettable sysvar ie, allow people to omit them from the accounts list
 // without breaking compat to be done after release, we keep the existing
 // interface for all instructions for compat with firedancer
+// JC: agreed!
 
 fn get_vote_state(vote_account_info: &AccountInfo) -> Result<Box<VoteState>, ProgramError> {
     if *vote_account_info.owner != solana_vote_program::id() {
         return Err(ProgramError::IncorrectProgramId);
     }
 
+    // JC: I think the code was added after this commit, but can we switch to
+    // `deserialize_into_uninit` to avoid the overhead of creating a
+    // `VoteState::default`?
     let mut vote_state = Box::<VoteState>::default();
     VoteState::deserialize_into(&vote_account_info.data.borrow(), &mut vote_state)
         .map_err(|_| ProgramError::InvalidAccountData)?;
```

Comment:
yup that is new!

(im going to go through and put comments on anything that requires it, then write my own pr and close this one and we can un-backwards the code review process. assume anything i dont comment on, im agreeing with)